### PR TITLE
feat(): add UI notification for client and HTTP errors

### PR DESCRIPTION
**Pull request by hhubik**
@timdeschryver any thoughts on this?  I tried to keep it simple: just extended the default error handler to show a snackbar and then hand the error off to the default error handler for logging.  Also added a simple test of localStorage availability.

### CODE REVIEW

**Comment by timdeschryver** [comment]
Great, thanks in advance! 
I'll take a look during the week.

File: src/app/app.component.ts [hunk 1]
```diff
@@ -169,4 +171,18 @@ export class AppComponent implements OnInit, OnDestroy {
...
     });
   }
+
+  /** Verifies that localStorage can be written to and read from */
+  private testLocalStorage() {
+    const testValue = 'testValue';
+    const testKey = 'testKey';
+    let retrievedValue: string;
+
+    localStorage.setItem(testKey, testValue);
```
Comment (timdeschryver) [inline src/app/app.component.ts]: I think adding wrapping this in a try-catch would be safer?
Also, should we also test if `localStorage` exists?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -169,4 +171,18 @@ export class AppComponent implements OnInit, OnDestroy {
...
+    retrievedValue = localStorage.getItem(testKey);
+
+    if (!retrievedValue || retrievedValue !== testValue) {
```
Comment (timdeschryver) [inline src/app/app.component.ts]: We can leave remove the `!retrievedValue` check here, because it's redundant.

**Comment by timdeschryver** [comment]
This is how it looks like for the ones wondering:

image

File: src/app/core/http-interceptors/http-error.interceptor.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+@Injectable()
+export class HttpErrorInterceptor implements HttpInterceptor {
+  constructor(private injector: Injector) {}
+
+  intercept(
+    request: HttpRequest<any>,
+    next: HttpHandler
+  ): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+      tap(null, httpErrorResponse => {
```
[the remark this answers is not in the document]
Comment (tomastrajan) [reply src/app/core/http-interceptors/http-error.interceptor.ts]: @timdeschryver the next step  would be `NotificationService` in core which will display styled toasts with `info`, `warn`, `error` API ;)

File: src/app/app.component.ts (hunk 1 above)
Comment (hhubik) [reply src/app/app.component.ts]: I will add the test of existence of `localStorage`.  As to wrapping this in `try-catch`, that seems to defeat the purpose of a global error handler.  What would we do in the `catch` besides re-throwing the exception?

Comment (tomastrajan) [reply src/app/app.component.ts]: Maybe this could be in the `core/local-storage.service` ?

Comment (hhubik) [reply src/app/app.component.ts]: @tomastrajan will do!

File: src/app/core/http-interceptors/http-error.interceptor.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (timdeschryver) [reply src/app/core/http-interceptors/http-error.interceptor.ts]: I think we can leave it as it is, with the comment @tomastrajan made.

What I meant was, doing something like this: 

```ts
 return next.handle(request).pipe(
      catchError(err => {
       ... handle error here...
      })
    );
```

**Comment by hhubik** [comment]
Sorry, I could not figure out how to sync my fork and my local repo with the starter project.  I am closing this and will open a new PR against a new fork.

**Pull request closed without merging**